mlflow/pyspark/ml/__init__.py

```diff
@@ -98,10 +98,17 @@ def _get_warning_msg_for_skip_log_model(model):
 
 def _should_log_model(spark_model):
     from pyspark.ml.base import Model
+    import re
 
     # TODO: Handle PipelineModel/CrossValidatorModel/TrainValidationSplitModel
     class_name = _get_fully_qualified_class_name(spark_model)
-    if class_name in _log_model_allowlist:
+    should_log = class_name in _log_model_allowlist
+    if not should_log:
+        for class_names in _log_model_allowlist:
+            if re.search(class_names, class_name):
+                should_log = True
+                break
```

Comment:
Let's improve the logic by:

1. check whether the item in the allowlist ends with "*"
2. if ends with "*", then compare it via "startswith"
3. otherwise compare it with "==".

Reply:
In that case do we only want to support the format of one trailing "*"?